rust/optify/src/builder/builder_trait.rs

```diff
@@ -0,0 +1,15 @@
+use std::path::Path;
+
+use crate::provider::OptionsProviderTrait;
+
+/// Trait defining the core functionality for building an options provider.
+///
+/// ⚠️ Development in progress ⚠️\
+/// Not truly considered public and mainly available to support bindings for other languages.
```

Comment:
You can add `#[doc(hidden)]` to hide it from API docs if you want

Reply:
Good to know. Would that hide it from https://docs.rs/optify ? Hmm... I think it might be weird to have it with no docs. Also, you need to import the trait to use certain methods on the builder. I needed to do that in tests. I'll clarify some of these warning messages.